### DISCUSSION TITLE
Strange behavior : removeTrait and addTrait are abruptly behaving dynamically

### DISCUSSION THREAD

**Discussion by codingmachine16** [GrapesJS/grapesjs#3995] [Q&A]
Hi @artf 

     on an event I am adding traits for select box as 

  e.addTrait(
    {
      label: "Categories",
      name: "sourceCategoriesAttr",
      id: "sourceCategoriesAttr",
      type: "select",
      options: [{name:"addidas", value:"addidas"},{name:"Nike","value:Nike"},{name:"Reboke",value: "Reboke"}],
      changeProp: 1,
    },

    { at: 3 }
  );
and on another event removing Traits like below

    e.removeTrait("sourceCategoriesAttr");

It is working fine upto here. But when add Traits again on the first event, it keep previous value selected. How to get rid of selected value.

 Please help.

Regards

**Comment by artf** [comment]
Traits read the value from the component, so if you need to change that value, change it at the component level, eg. `component.set('sourceCategoriesAttr', '')`
